src/narjure/memory_management/persistence_manager.clj

```diff
@@ -2,18 +2,47 @@
   (:require
     [co.paralleluniverse.pulsar.actors :refer [self ! whereis cast! Server gen-server register! shutdown! unregister! set-state! state]]
     [narjure.actor.utils :refer [defactor]]
+    [narjure.memory-management.task-dispatcher :refer [c-map]]
+    [narjure.memory-management.concept-creator :refer [create-concept]]
+    [narjure.actor.utils :refer [read-seq-from-file]]
     [taoensso.timbre :refer [debug info]])
   (:refer-clojure :exclude [promise await]))
 
 (def aname :persistence-manager)
 
-(defn concept-state-request-handler
-  "Processes :concept-state-request-msg:
-    serialises concept state and returns
-    to requesting actor"
-  [from [msg]]
-  ()
-  (debug aname "process-concept-state-request-msg"))
+(defn persist-concept-state-handler
+  "Posts :concept-state-request-msg to each concept in c-map"
+  [from [_ path]]
+  (set-state! (assoc @state :path path :concept-count (count @c-map) :received-states 0))
+  (doseq [concept-ref (vals @c-map)]
+    (cast! concept-ref [:concept-state-request-msg]))
+  #_(debug aname "process-persist-concept-state-msg"))
+
+(defn concept-state-handler
+  "process each :concept-state-msg by serialising the state to backing store.
+   where state specifies the path of the backing store. The number of received
+   states is tracked. The file is overwritten."
+  [from [_ concept-state]]
+  (let [c-state (dissoc concept-state :active-concept-collator :general-inferencer :forgettable-concept-collator)]
```

Comment:
now it is
```
(defn concept-state-handler
  "process each :concept-state-msg by serialising the state to backing store.
   where state specifies the path of the backing store. The number of received
   states is tracked. The file is overwritten."
  [from [_ concept-state]]
  (let [c-state (dissoc concept-state :active-concept-collator :general-inferencer :forgettable-concept-collator)]
    (if (= (:received-states @state) 0)
      (spit (:path @state) c-state)
      (spit (:path @state) c-state :append true))
    )
  (set-state! (update @state :received-states inc))
  (when (= (:received-states @state) (:concept-count @state))
    (info aname "Persisting concept state to disk complete")))
```
but it can be 
```
(defn concept-state-handler
  "process each :concept-state-msg by serialising the state to backing store.
   where state specifies the path of the backing store. The number of received
   states is tracked. The file is overwritten."
  [_ [_ concept-state]]
  (let [c-state (dissoc concept-state
                        :active-concept-collator
                        :general-inferencer
                        :forgettable-concept-collator)
        {:keys [path received-states concept-count]} @state]
    (if (zero? received-states)
      (spit path c-state)
      (spit path c-state :append true))
    (set-state! (update @state :received-states inc))
    (when (= (:received-states @state) concept-count)
      (info aname "Persisting concept state to disk complete"))))
```

The idea is that you can fetch all keys of collection in one place, instead of writing `(:key colllection)` each time, especially when you have to write it more than once, like `(:path @state)`.